spinnaker-monitoring-daemon/spinnaker-monitoring/newrelic_service.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2019 New Relic Corporation. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import spectator_client
+import logging
+import os
+from newrelic_telemetry_sdk import CountMetric, GaugeMetric, MetricClient
+
+
+class NewRelicMetricsService(object):
+    """A metrics service for interacting with New Relic."""
+
+    def __init__(self, spectator_helper, metric_client, tags, options):
+        self.spectator_helper = spectator_helper
+        self.metric_client = metric_client
+        self.tags = tags
+
+    def parse_metric(self, service, metric_name, metric_instance, metric_data, service_data, metric_list):
+        kind = self.spectator_helper.determine_primitive_kind(
+            metric_data["kind"])
+        for metric_data_value in metric_instance["values"]:
+            tags = self.tags.copy()
+            for tag in metric_instance["tags"]:
+                tags[tag["key"]] = tag["value"]
+            tags["applicationName"] = service_data["applicationName"]
+            tags["applicationVersion"] = service_data["applicationVersion"]
+            interval = metric_data_value["t"] - \
+                service_data["__collectStartTime"]
```

Comment:
I dont think you want this as your interval. collectStartTime is when you started collecting the data. "t" is the time spinnaker recorded the data. I presume the "interval" that the API wants is the time period that the count was accumulated over, which would be the start time of the process providing the metric, service_metadata['startTime'].
So interval might be metric_data_value['t'] - service_metadata['startTime']. Hopefully both are in the same time units, which is probably epoch milliseconds. Otherwise there might be a conversion involved.

Reply:
Thanks, this looks like a legit bug. Would be happy to open a second PR with fixes for this.